kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -342,6 +382,23 @@ private Map<String, String> getOperationParameters() {
     return Collections.emptyMap();
   }
 
+  private void recordTransactionReport(
+      Engine engine,
+      Optional<Long> committedVersion,
+      TransactionMetrics transactionMetrics,
+      Optional<Exception> exception) {
+    TransactionReport transactionReport =
+        new TransactionReportImpl(
+            dataPath.toString() /* tablePath */,
+            operation.toString(),
+            engineInfo,
+            committedVersion,
+            transactionMetrics,
+            readSnapshot.getSnapshotReport(),
```

Comment:
what is the readSnapshot of a new table creation? is it just `InitialSnapshot`? with version = -1?

Reply:
Yeah we create it here https://github.com/delta-io/delta/blob/master/kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java#L114